kmp-library/src/iosMain/kotlin/com/novoda/aws/v4/signer/hash/Hmac256Encoder.kt

```diff
@@ -0,0 +1,24 @@
+package com.novoda.aws.v4.signer.hash
+
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.convert
+import kotlinx.cinterop.usePinned
+import platform.CoreCrypto.CCHmac
+import platform.CoreCrypto.CC_SHA256_DIGEST_LENGTH
+import platform.CoreCrypto.kCCHmacAlgSHA256
+
+actual object Hmac256Encoder {
```

Comment:
This is the native implementation using native c libraries under `platform.CoreCrypto` through [object pinning](https://blog.jetbrains.com/kotlin/2017/11/kotlinnative-v0-4-released-objective-c-interop-webassembly-and-more/).